### ISSUE TITLE
[Problem/Bug]: When using a custom scheme, opening a secondary window fails

### ISSUE THREAD

**Issue by reynoldslabware** [MicrosoftEdge/WebView2Feedback#4085]
### What happened?

Our application uses a custom protocol (labware://labware).
Our first window loads successfully (bootstrap.html).
The top and bottom of the main console are shown below.
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/143733690/dcf8e5d2-4a68-41f8-b4c3-a2cac2321b50)

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/143733690/4a7d889d-ebba-4feb-b849-63e70f3edbe3)

Our application then opens a second page using 
`window.open("whatever", "_blank")
`
That task page result is just grey for some reason. See the red error text in the console above.
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/143733690/9f85f849-1a21-4cb6-82c8-0891d33e3cd5)

It is odd that the tooltip for the main console log messages shows the custom scheme, things got loaded,
but the attempt to open another page seems to complain about the lack of a custom scheme
that seemed to have just worked.

In my support DLL which interfaces to WebView2 I do see logging entries for the main bootstrap page such as,
```
Navigate labware://labware/lwweb/bootstrap.html
myNavigationStartingEventHandler() called
WebResourceRequested labware://labware/lwweb/bootstrap.html
WebResourceRequested labware://labware/lwweb/css/common/lw-theme/jquery-ui-1.12.1.custom.css
```
I do not see such any such entries for the second page.
Any direction on what I should be looking at would be appreciated.

### Importance

Important. My app's user experience is significantly compromised.

### Runtime Channel

Prerelease (Edge Canary/Dev/Beta)

### Runtime Version

117.0.2045.31

### Framework

Win32

### Operating System

Windows 10

### Repro steps

This can be reproduced when using a custom schema and loading this HTML page

```
<html>
<script>
function test2()
{
 window.open("another.html", "_blank");
}
</script>
<body>
 <button onclick="test2()">Test2</button>
</body>
</html>
```
With the another.html file containing
```
<html>
<body>
Hello world
</body>
</html>

```

### Regression

No, this never worked

**Comment by lflores-ms** [comment]
@yildirimcagri-msft this looks like a bug?

**Comment by yildirimcagri-msft** [comment]
This seems to be because the opened window is opened as a basic popup window and is not associated to a WebView2 and as a result the `WebResourceRequest` with the custom scheme cannot be handled by the app. For this to work, you'd need to associate a new WebView2 to the opened window by handling the [NewWindowRequested event](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.newwindowrequested?view=webview2-dotnet-1.0.2045.28) and setting `NewWindow` and then you can set a `WebResourceRequested` event handler on the new WebView2.

**Comment by reynoldslabware** [comment]
Thank you for the guidance. It sounds like the high level steps to get this to cooperate
in the NewWindowRequestedEventHandler,
  HRESULT myNewWindowRequestedEventHandler(
    ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args)
are:

 - get t(he deferal
    args->GetDeferral(&deferral);

 - create a new web view controller
  HRESULT cwvRc = m_env->CreateCoreWebView2Controller(hWndParent, 
    Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(...)
    )
What should the parent window be?
I guess the originating web view's parent because the originating web view itself 
may close independently of the about to be created web view.

 - use that controller to create a new web view

 - setup the new web view's callbacks including the WebResourceRequested callback

 - adjust the new web view's position and size as desired

 - adjust the initiating NewWindowRequestedEventArgs argument via
    args->put_NewWindow(newWebView);
    args->put_Handled(TRUE);

 - move the original NewWindowRequestedEventHandler call onward via
    deferral->Complete();

**Comment by reynoldslabware** [comment]
In CEF3 when we did a window.open() call a new Window was automatically created and the referenced URI content was shown within it. That new Window mapped logically to a new tab in a regular browser.
My approach above was close. 
The parent window value above needs rethinking because I ended up with 2 web views in my main window. Oops.
Ay thoughts on what the CEF3 world was doing for me automatically?
